### PR TITLE
Issue 563 parameters by material

### LINKED ISSUE #563: add parameters by material
add parameters by choosing materials e.g. `pybamm.ParameterValues(cathode=...,anode=...)` then have library which allows user to pick and choose different materials

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

- [x] Reorganised parameters by material.
- [x] Specified path to functions in csv files
- [x] Allowed setting either current or C-rate
- [x] Updated docs on how to add new parameters

Fixes #563 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by rtimms** [comment]
some comments for discussion:

- I think there could be a "geometry" or "cell" file which contains the things like cell geometry and maybe other cell-scale parameters like voltage cut-off or number of cells to make a battery. Maybe nominal cell capacity as stated by manufacturer. From this we then define the typical current to be 1C based on the nominal cell capacity.

- For li-ion we will need a set of thermal parameters. The structure could be the same for lead-acid, with a boring file with the ref and max temp in

- It would be good to have the ability to just pass the name of a paper to quickly set all the parameters from that paper 

- I think a misc params file is probably ok, but should also be named with author_year 

- maybe not for this issue, but for effective properties e.g. conductivity it would be nice to select how they are calculated e.g. bruggeman or otherwise. At the moment we hard-code bruggeman into the models

- I think this has already been discussed, but where is the best place to set initial conditions?

**Comment by ferranbrosa** [comment]
I like @Scottmar93 suggestion in #563 about defining an experiment. It is good to distinguish between the following: battery parameters (intrinsic of the battery materials or, from a maths point of view, of the equations), which could be split into further subcategories; geometry and those related to the experiment (how we cycle the battery). This poses some interesting questions about defining the SoC of the electrodes (or maybe we should use state of lithiation to avoid confusion) as they are related to the potentials. 

I agree also with @rtimms points above. In particular, I think that we should define a new parameter/function that accounts for the effectiveness of the medium, so then one can choose between different models. In particular, I think it would be interesting to use the values obtained via homogenisation using Rayleigh's multipole method (see Bruna & Chapman 2015).

File: docs/tutorials/add-parameter-values.rst [hunk 1]
```diff
@@ -16,32 +16,61 @@ At the stage of creating a model, we use :class:`pybamm.Parameter` and :class:`p
...
+
+    param = pybamm.ParameterValues(
+        chemistry={
+            "chemistry": "lithium-ion",
+            "cell": "kokam_Marquis2019",
+            "anode": "new_anode_chemistry_Bloggs2019",
+            "separator": "separator_Marquis2019",
+            "cathode": "lico2_Marquis2019",
+            "electrolyte": "lipf6_Marquis2019",
+            "experiment": "1C_discharge_from_full_Marquis2019",
```
Comment (rtimms) [inline docs/tutorials/add-parameter-values.rst]: Maybe there should be a very brief description of the parameters that go into the experiment folder? Though maybe it is clear from looking at the existing parameter sets.

File: pybamm/parameters/parameter_sets.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#
+# Parameter sets from papers
```
Comment (rtimms) [inline pybamm/parameters/parameter_sets.py]: this is very user-friendly!

File: docs/tutorials/add-parameter-values.rst (hunk 1 above)
Comment (ferranbrosa) [reply docs/tutorials/add-parameter-values.rst]: I also think that a list of the required parameters in the documentation would be helpful.

**Pull request merged**